case-lib/pipeline.sh

```diff
@@ -2,6 +2,15 @@
 
 func_pipeline_export()
 {
+    # For legacy HDA platform, there is no topology, we have to exporting pipeline
+    # information from proc file system.
+    is_sof_used || {
+        dlogi "No SOF sound card found, exporting pipeline parameters from proc file system"
+        source <(sof-dump-status.py -e) ||
+            die "Failed to export pipeline parameters from proc file system"
+        return 0
```

Comment:
I would rather drop `return 0`. In the very unlikely case `source` fails then it's better not to return 0.

I just realize one drawback with this `source <(   )` solution: it can probably never fail. I mean even if `sof-dump-status.py` is not found or fails, `source <(   )` will still succeed. I hope most `sof-dump-status.py` error message / stack traces get at least printed to `stderr` so they will "escape" `< ( )` and be visible in the CI logs.

If you are not too worried about `sof-dump-status.py` failures then you can keep `source <( )`. If you are worried about some potential `sof-dump-status.py` failures then you should source an intermediate `mktemp` file instead of using `< ( )`

Reply:
I think we should care the sof-dump-status.py failures, it can help to detect system/sof-test issues. A intermediate temp file is used, and error handler is added.
The `return` is required, because we don't want the below code(tplg pipeline exporting) to be executed.

Reply:
> The return is required, because we don't want the below code(tplg pipeline exporting) to be executed.

Ooops, sorry for the noise.

> A intermediate temp file is used

OK, then please use a `mktemp` (and delete it). I realize we don't support concurrency for now but we may want to in the future and this code is in a very generic and common function.